src/min-reviewers.ts

```diff
@@ -0,0 +1,101 @@
+import { ReviewersAll, Label, PullContext, Review } from "./types";
+import * as core from "@actions/core";
+import * as github from "@actions/github";
+
+export function getMinReviewers(labels: Label[]): ReviewersAll | number {
+  const pattern = /min-(?<number>\d|all)-reviewers/;
+
+  for (const label of labels) {
+    const m = label.name.match(pattern);
+    if (!m) {
+      continue;
+    }
+
+    const { groups = {} } = m;
+    const { number = "0" } = groups;
+    return number === "all" ? number : parseInt(number, 10);
+  }
+
+  return 0;
+}
```

Comment:
If we only have one match here you could ditch number group, simplify regex and just check what is the first match if it exists:
```
export function getMinReviewers(labels: Label[]): ReviewersAll | number {
  const pattern = /min-(\d|all)-reviewers/;

  for (const label of labels) {
    const m = label.name.match(pattern);

    if (m) {
      return m[1] === 'all' ? m[1] : parseInt(m[1], 10);
    }
  }

  return 0;
}
```

Reply:
Also this maybe a bit of a stretch, but wont this code return the first label that it will find that matches the pattern and ignore the others, so if labels = ['min-1-reviewers', 'min-all-reviewers'] only 1 reviewer will be required.

Reply:
It will always take the first matched label, there is even a test for it :)